interpreter/cling/lib/Interpreter/DeclCollector.cpp

```diff
@@ -321,4 +321,21 @@ namespace cling {
     m_Consumer->HandleCXXStaticMemberVarInstantiation(D);
   }
 
+  void DeclCollector::DeclRead(clang::serialization::DeclID, const clang::Decl *D) {
+    assertHasTransaction(m_CurTransaction);
+
+    assert(D & "Decl doesn't exist!");
+    if (!D->hasOwningModule()) return;
+
+    clang::Module *M = D->getOwningModule();
+    assert(M & "Could not get clang::Module!");
```

Comment:
That assert seems dead code (and using bitwise and). I do not see how `M` will be 0 given the early return.